tsconfig.json

```diff
@@ -1,5 +1,20 @@
-// Yes this file is intentionally empty!
+// Yes this file is intentionally near-empty!
 // ---
 // Having a blank `tsconfig.json` file prevents TypeScript from crawling up your directory tree
-// and possibly picking up a parent `tsconfig.json` (which, unsurprisingly, is very hard to debug)
-{}
+// and possibly picking up a parent `tsconfig.json` (which, unsurprisingly, is very hard to debug).
+// This is also used for linting out-of-project files with type information.
+{
+  "extends": "./tsconfig.base.json",
+  "compilerOptions": {
+    "allowJs": true,
+    "noEmit": true
+  },
+  "include": [
+    ".",
+    "packages/upgrade/upgrade.mjs",
+    // ...and yet:
+    // /Users/josh/repos/astro/packages/upgrade/upgrade.mjs
+    // 0:0  error  Parsing error: /Users/josh/repos/astro/packages/upgrade/upgrade.mjs was not found by the project service.
+    // Consider either including it in the tsconfig.json or including it in allowDefaultProject
+  ]
```

Comment:
Well, I figured it out. https://github.com/JoshuaKGoldberg/repros/tree/tseslint-project-service-non-included-nearest-tsconfig is a simplified reproduction.

Basically:
* `packages/upgrade/tsconfig.json` includes `["src"]`
* `packages/upgrade/upgrade.mjs` is not in that includes
* The project service therefore isn't picking up the file

This would be a good case to stick with `parserOptions.project`, to give more fine-grained control over what projects include each file. I filed https://github.com/typescript-eslint/typescript-eslint/issues/9753 to document this better.